xarray/tests/test_dataarray.py

```diff
@@ -3140,18 +3140,22 @@ def test_series_categorical_index(self):
         arr = DataArray(s)
         assert "'a'" in repr(arr)  # should not error
 
-    def test_to_and_from_dict(self):
+    @pytest.mark.parametrize("encoding", [True, False])
+    def test_to_and_from_dict(self, encoding):
```

Comment:
```suggestion
    def test_to_and_from_dict(self, encoding) -> None:
```
Maybe mypy will find something?